openfeature/client.py

```diff
@@ -361,19 +361,13 @@ def evaluate_flag_details(  # noqa: PLR0915
             return flag_evaluation
 
         except OpenFeatureError as err:
-            logger.exception(
-                "Error %s while evaluating flag with key: '%s'",
-                err.error_code,
-                flag_key,
-            )
-
             error_hooks(flag_type, hook_context, err, reversed_merged_hooks, hook_hints)
 
             return FlagEvaluationDetails(
                 flag_key=flag_key,
                 value=default_value,
                 reason=Reason.ERROR,
-                error_code=err.error_code,
+                error_code=err.error_code or ErrorCode.GENERAL,
```

Comment:
This shouldn't be needed, because every OF error has an error code.

Reply:
Okay, I can remove it. We just did something similar in JS and the typings implied that this was optional.